rust/processor/src/db/postgres/migrations/2025-01-08-171729_emojicoin_arena/up.sql

```diff
@@ -158,217 +169,40 @@ WITH melee AS (
-            start_time = NEW.start_time,
-            duration = NEW.duration,
-            max_match_percentage = NEW.max_match_percentage,
-            max_match_amount = NEW.max_match_amount
-        WHERE arena_info.melee_id = NEW.melee_id;
-        RETURN NEW;
-    END;
-$$ LANGUAGE plpgsql;
-
--- Since events can be inserted out of order, we need to handle both the case
--- where the Melee event is inserted first and where the Enter event is
--- inserted first.
-CREATE FUNCTION update_arena_info_enter() RETURNS trigger AS $$
-    BEGIN
-        INSERT INTO arena_info (
-            melee_id,
-            volume,
-            rewards_remaining,
-            apt_locked
-        ) VALUES (
-            NEW.melee_id,
-            NEW.quote_volume,
-            0 - NEW.match_amount,
-            NEW.quote_volume
-        )
-        ON CONFLICT (melee_id) DO
-        UPDATE SET
-            volume = arena_info.volume + NEW.quote_volume,
-            rewards_remaining = arena_info.rewards_remaining - NEW.match_amount,
-            apt_locked = arena_info.apt_locked + NEW.quote_volume
-        WHERE arena_info.melee_id = NEW.melee_id;
-        RETURN NEW;
-    END;
-$$ LANGUAGE plpgsql;
-
-CREATE FUNCTION update_arena_info_exit() RETURNS trigger AS $$
-    BEGIN
-        UPDATE arena_info SET
-            -- We check that this never underflows due to rounding errors
-            apt_locked = GREATEST(ROUND(
-                arena_info.apt_locked
-                    - NEW.emojicoin_0_proceeds
-                        / NEW.emojicoin_0_exchange_rate_base
-                        * NEW.emojicoin_0_exchange_rate_quote
-                    - NEW.emojicoin_1_proceeds
-                        / NEW.emojicoin_1_exchange_rate_base
-                        * NEW.emojicoin_1_exchange_rate_quote
-            ), 0)
-        WHERE arena_info.melee_id = NEW.melee_id;
-        RETURN NEW;
-    END;
-$$ LANGUAGE plpgsql;
-
-CREATE FUNCTION update_arena_info_swap() RETURNS trigger AS $$
-    BEGIN
-        UPDATE arena_info SET
-            volume = arena_info.volume + NEW.quote_volume
-        WHERE arena_info.melee_id = NEW.melee_id;
-        RETURN NEW;
-    END;
-$$ LANGUAGE plpgsql;
-
-CREATE TRIGGER create_melee_info_trigger AFTER INSERT ON arena_melee_events
-    FOR EACH ROW EXECUTE FUNCTION create_melee_info();
-
-CREATE TRIGGER update_arena_info_enter_trigger AFTER INSERT ON arena_enter_events
-    FOR EACH ROW EXECUTE FUNCTION update_arena_info_enter();
-
-CREATE TRIGGER update_arena_info_exit_trigger AFTER INSERT ON arena_exit_events
-    FOR EACH ROW EXECUTE FUNCTION update_arena_info_exit();
-
-CREATE TRIGGER update_arena_info_swap_trigger AFTER INSERT ON arena_swap_events
-    FOR EACH ROW EXECUTE FUNCTION update_arena_info_swap();
-
--- Indices
-
-CREATE INDEX latest_swap_by_market_address ON swap_events (market_address, market_nonce DESC);
+CREATE VIEW arena_leaderboard_history_with_arena_info AS
+SELECT
+    arena_leaderboard_history.melee_id,
```

Comment:
```suggestion
CREATE VIEW arena_leaderboard_history_with_arena_info AS
SELECT
    arena_leaderboard_history.user,
    arena_leaderboard_history.melee_id,
```

You need to include `user` here, I think you accidentally left it out because you converted it from the function, which passed it in as a parameter

Reply:
is there any reason this can't just be arena_leaderboard_history.* and arena_info.* ?

I think you only leave out 4 or 5 fields total, I think it might be simpler if you just join them entirely, especially if we add more data down the road to those views it auto updates

Reply:
I don't think it will be automatically added. You'd have to drop and recreate the view.

Reply:
Addressed in 0d1ed8d.

Reply:
Yeah you're right on that- didn't mean to write that authoritatively, it was supposed to be more of a question, haha, but I looked it up and see that you're right